src/hotspot/share/gc/g1/g1FullGCPrepareTask.cpp

```diff
@@ -168,9 +168,12 @@ size_t G1FullGCPrepareTask::G1PrepareCompactLiveClosure::apply(oop object) {
 size_t G1FullGCPrepareTask::G1RePrepareClosure::apply(oop obj) {
   // We only re-prepare objects forwarded within the current region, so
   // skip objects that are already forwarded to another region.
-  oop forwarded_to = obj->forwardee();
-  if (obj->is_forwarded() && !_current->is_in(forwarded_to)) {
-    return obj->size();
+  if (obj->is_forwarded()) {
```

Comment:
Yeah, this seems like a nice micro-optimization for upstream. Do you want to take care of it?

Reply:
Naa, this doesn't really do much, unless we extract the header once, and then check is_forwarded() on it, and avoid decoding if it's not forwarded - but even then this seems to be too micro ;-)
Here it is not an optimization but a necessary change, because we cannot blindly decode the forwardee anymore.